### DISCUSSION TITLE
Why is the output for--encoding=windows-1252 a UTF-8 encoded file?

### DISCUSSION THREAD

**Discussion by datatraveller1** [BurntSushi/ripgrep#1933] [General] [answered]
I'm wondering why the output file encoding is UTF-8 when I use --encoding=windows-1252:
rg "théâtre" --encoding=windows-1252 in.txt > out.txt

in.txt uses codepage 1252 but out.txt gets UTF-8 encoded.
If I use an encoding for searching I want the same encoding for the output.

Ok, other people might consider this rather a feature than a bug so I suggest to add a `--out-encoding` option.

**Comment by BurntSushi** [comment, marked as answer]
Always emitting UTF-8 is a simplification for the implementation. But more generally, the only reason ripgrep is encoding aware at all is to support automatic searching of UTF-16. The other encodings that come from encoding_rs are just a bonus. The main issue is that it is difficult to do this outside of ripgrep.

But ripgrep always emits utf8 (or raw bytes if no encoding is used on the input) and only prints to stdout. So it's trivial to pipe it through your transcoding program of choice.

**Reply by datatraveller1** [reply]
ok, so I can use something like
rg "théâtre" --encoding=windows-1252 in.txt | iconv -c -s -f UTF-8 -t CP1252 > out.txt

**Reply by datatraveller1** [reply]
I tried to change this discussion to "Answered" but couldn't succeed. Can you do it for me, please?

**Reply by datatraveller1** [reply]
Thank you again for the solution. I have just got one thought:
**xsv** has an option:
`-o, --output <file>    Write output to <file> instead of stdout.`
_In my experience this keeps the current encoding._
Would this be maybe also useful for ripgrep?

**Reply by BurntSushi** [reply]
Sorry, but I'm not adding a `--output` flag or an option to specifically write the output in an encoding other than UTF-8.

**Reply by BurntSushi** [reply]
And a `-o/--output` flag seems totally orthogonal to encoding.

**Reply by datatraveller1** [reply]
I don't think `--output` is orthogonal to encoding. In fact, in combination with an `--out-encoding` option these two options would be the only proper way to use an out encoding on Windows (without the need of additional software like iconv). But no problem, I can use the solution above and I think an `--output`  option kind of contradicts the concept of a grep program where usually printing to stdout (and if needed redirecting stdout into a file) is used. So thank you again! Both ripgrep and xsv are great programs.
